### PR TITLE
Got rid of the old style std/ includes for several files, mostly in map/ and platform/.

### PR DESCRIPTION
**Pull request by mpimenov**

### CODE REVIEW

**Comment by mpimenov** [comment]
JTALL

File: search/house_numbers_matcher.cpp [hunk 1]
```diff
@@ -4,13 +4,13 @@
 
 #include "base/logging.hpp"
 
+#include "std/transform_iterator.hpp"
```
Comment (tatiana-yan) [inline search/house_numbers_matcher.cpp]: может тоже менять на 
```
#include <boost/iterator/transform_iterator.hpp>
```
?

Comment (mpimenov) [reply search/house_numbers_matcher.cpp]: done

**Comment by tatiana-yan** [comment]
```
In file included from /home/jenkins/jenkins-root/workspace/LinuxGCC/omim/map/displacement_mode_manager.cpp:1:0:
/home/jenkins/jenkins-root/workspace/LinuxGCC/omim/map/displacement_mode_manager.hpp:38:48: error: 'CHAR_BIT' was not declared in this scope
   static_assert(SLOT_COUNT <= sizeof(m_mask) * CHAR_BIT, "Number of slots is too large");
```

**Comment by mpimenov** [comment]
JTLINUX

**Pull request merged**